File: fetch_table.scala
import org.apache.spark.sql.SparkSession

object FetchTableStructure {
  def main(args: Array[String]): Unit = {

    // Define connection parameters
    val tenantId = "<your-tenant-id>"
    val clientId = "<your-client-id>"
    val clientSecret = "<your-client-secret>"
    val synapseServer = "<your-synapse-server>"
    val database = "<your-database-name>"
    val tableName = "your_table"

    // Initialize SynapseConnection
    val synapseConn = new SynapseConnection(tenantId, clientId, clientSecret, synapseServer, database)

    // Create Spark session
    val spark = SparkSession.builder()
      .appName("SynapseConnect")
      .getOrCreate()

    // Query to fetch table schema
    val schemaQuery = s"(SELECT COLUMN_NAME, DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_NAME = '$tableName') as schema_table"

    // Read schema from Synapse
    val df = spark.read
      .format("jdbc")
      .option("url", synapseConn.getJdbcUrl())
      .option("dbtable", schemaQuery)
      .options(synapseConn.getConnectionProperties())
      .load()

    // Show table structure
    df.show()
  }
}

File: fetch_table.py
from pyspark.sql import SparkSession
from SynapseConnection import SynapseConnection

# Initialize connection details
tenant_id = "<your-tenant-id>"
client_id = "<your-client-id>"
client_secret = "<your-client-secret>"
synapse_server = "<your-synapse-server>"
database = "<your-database-name>"

# Create a SynapseConnection instance
synapse_conn = SynapseConnection(tenant_id, client_id, client_secret, synapse_server, database)

# Initialize Spark session
spark = SparkSession.builder.appName("SynapseConnect").getOrCreate()

# Fetch table structure
table_name = "your_table"
df = spark.read \
    .format("jdbc") \
    .option("url", synapse_conn.get_jdbc_url()) \
    .option("dbtable", f"(SELECT COLUMN_NAME, DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_NAME = '{table_name}') as schema_table") \
    .options(**synapse_conn.get_connection_properties()) \
    .load()

# Show table structure
df.show()
